sdk/core/Azure.Core/tests/TestFramework/RecordMatcher.cs

```diff
@@ -171,26 +178,47 @@ private string JoinHeaderValues(string[] values)
             return string.Join(",", values);
         }
 
+        private string[] RenormalizeSemicolons(string[] values)
+        {
+            string[] outputs = new string[values.Length];
+            for (int i = 0; i < values.Length; i++)
+            {
+                outputs[i] = string.Join("; ", values[i].Split(';').Select(part => part.Trim()));
+            }
+
+            return outputs;
+        }
+
         private int CompareHeaderDictionaries(SortedDictionary<string, string[]> headers, SortedDictionary<string, string[]> entryHeaders, HashSet<string> ignoredHeaders, StringBuilder descriptionBuilder = null)
         {
             int difference = 0;
             var remaining = new SortedDictionary<string, string[]>(entryHeaders, entryHeaders.Comparer);
             foreach (KeyValuePair<string, string[]> header in headers)
             {
-                if (remaining.TryGetValue(header.Key, out string[] values))
+                var requestHeaderValues = header.Value;
+                var headerName = header.Key;
+
```

Comment:
Could we just `continue` here if `ignoredHeaders.Contains(headerName)` and remove the two checks for this below?

Reply:
Let me see, you might be right.